### DISCUSSION TITLE
Pydantic v2: `RequestValidationError` just contains text instead of structured error details

### DISCUSSION THREAD

**Discussion by chbndrhnns** [fastapi/fastapi#10338] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

import fastapi
import pydantic
from fastapi.exceptions import RequestValidationError
from starlette.responses import JSONResponse
from starlette.testclient import TestClient


async def handle_err(request, exc):
    return JSONResponse(content={"details": exc.args[0]}, status_code=422)

app = fastapi.FastAPI(exception_handlers={pydantic.ValidationError: handle_err,RequestValidationError: handle_err,})


class Pagination(pydantic.BaseModel):
    skip: int = 0
    limit: Annotated[int, pydantic.Field(ge=1)] = 1


@app.get("/")
async def get(pagination: Pagination = fastapi.Depends()):
    return {"data": pagination}


def test_():
    with TestClient(app) as client:
        res = client.get("/?limit=-1")
        assert res.json()["details"] == [{'ctx': {'limit_value': 1},
  'loc': ['query', 'limit'],
  'msg': 'ensure this value is greater than or equal to 1',
  'type': 'value_error.number.not_ge'}]
```

### Description

* Run test with pydantic v2: fails
* Run test with pydantic v1: passes

This means it is no longer possible to extract details of an error without parsing text. 

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.4.0

### Python Version

3.9.15

**Comment by YuriiMotov** [comment, marked as answer]
https://fastapi.tiangolo.com/tutorial/handling-errors/#override-the-httpexception-error-handler
